### DISCUSSION TITLE
WebSocket - how to disable compression

### DISCUSSION THREAD

**Discussion by lzalewsk** [reflex-dev/reflex#2070] [Q&A]
Hi, Is there any way to disable compression for web socket messages during reflex run (precisely  for FastAPI backend)?
WebSocket disconnects when authentication is enabled within Azure container App service. More in [this issue](https://github.com/microsoft/azure-container-apps/issues/812). 
I am wandering how to add ws_per_message_deflate=False flag
Any suggestions?

**Comment by masenf** [comment]
We don't currently support this, nor does reflex even expose the configuration where this could be overridden.

I have made a small change to allow the user to override the `gunicorn_worker_class`, which is used in the production server: https://github.com/reflex-dev/reflex/pull/2084

With this change in place, I'm able to add the following to my `rxconfig.py` to disable the config knob you mentioned:

```python
import uvicorn.workers

import reflex as rx


class NoWSPerMessageDeflate(uvicorn.workers.UvicornH11Worker):
    CONFIG_KWARGS = {
        **uvicorn.workers.UvicornH11Worker.CONFIG_KWARGS,
        "ws_per_message_deflate": False,
    }


config = rx.Config(
    app_name="repro_bool_state",
    gunicorn_worker_class="rxconfig.NoWSPerMessageDeflate",
)
```

This does only work in prod mode `--env prod`, since in dev mode we just use `uvicorn.run` directly (and not gunicorn). But i think that's probably okay.

**Reply by masenf** [reply]
@lzalewsk This change merged in reflex-0.3.2, so you can give it a try and see if it has the intended effect!
